lib/pharos/cluster_manager.rb

```diff
@@ -44,10 +44,15 @@ def load
       Pharos::HostConfigManager.load_configs(@config)
     end
 
+    def gather_facts
+      apply_phase(Phases::GatherFacts, config.hosts, ssh: true, parallel: true)
+    end
+
     def validate
       addon_manager.validate
+      gather_facts
       apply_phase(Phases::ValidateHost, config.hosts, ssh: true, parallel: true)
-      apply_phase(Phases::ValidateHostname, config.hosts, ssh: false, parallel: false)
+      apply_phase(Phases::ValidateClusterHosts, config.hosts, ssh: false, parallel: false)
```

Comment:
If we `GatherFacts` first to get the `@host.hostname`, then the `ValidateHostname` logic can also be done in `ValidateHost`.

Reply:
IOW, the reason for the separate `ValidateHostClusters` phase was that the hostnames were gathered in parallel as part of `ValidateHost`... so there's no more need to separate the two validation phases anymore.

Reply:
Good catch, I'll refactor this more.